### ISSUE TITLE
Cannot Compile Sample Pawn Code

### ISSUE THREAD

**Issue by CJRobertson** [WhoLsJohnGalt/watson-unreal-sdk-plugin#12]
I had followed the directions in the Read Me, but unfortunately I can't get any of the sample code to compile for a launcher version of 4.18.3 (have not tried this with a source build of the engine).

Before compiling the project I'm left with the MyPawn sources files having a lot of undefined functions even though they are clearly there and at least to the best of my knowledge is setup correctly. On top of that it seems that Visual Studio doesn't think the AMyPawn class is valid.

**MyPawn.h**
![h](https://user-images.githubusercontent.com/13307644/36348843-3af3c4f0-142d-11e8-8f74-6abdc949801f.PNG)

**MyPawn.cpp**
![cpp](https://user-images.githubusercontent.com/13307644/36348841-340b91a4-142d-11e8-90fb-b76dd134457f.PNG)

After attempting to compile the project I get what you see in the provided output log which is apparently I'm missing a UCLASS name on line one of the MyPawn.h file, though that is taken care of on line ten it looks like. Compiling the project with the plugin but WITHOUT the MyPawn source file changes always ends in a successful compile so I don't think the plugin is causing any of this.

**Output Log:**
[VS_FailedOutputLog.txt](https://github.com/Perefin/watson-unreal-sdk-plugin/files/1734441/VS_FailedOutputLog.txt)

I'm not really sure how to solve this issue sadly :/

**Comment by AJDomi** [comment]
Hello, I will try to help you find the issue. One thing I notice is the macro between class and AMyPawn should probably be IBMWATSONTEST_API with no <> around it. If it is correct, it should be purple in color.
Let me know if that works. If it does not we can continue troubleshooting from there.

**Comment by CJRobertson** [comment]
Getting closer after that suggestion. The MyPawn.cpp file still looks the same with all the errors, but the MyPawn.h file is still having an issue or two (just not the same ones). Apparently VS is looking for a ; at the end of the class declaration, but it seems the semicolon is there already. I've attached the output log and a screenshot of the .h file.

**VS Output Log:**
[VSOutputLog.txt](https://github.com/Perefin/watson-unreal-sdk-plugin/files/1745650/VSOutputLog.txt)

![almostfixedhfile](https://user-images.githubusercontent.com/13307644/36504427-5278a890-1705-11e8-88e5-3828f6a19302.PNG)

**Comment by AJDomi** [comment]
The issue is still caused by the same thing. IBMWatsonTest_API is not the macro that should be there. It has to be IBMWATSONTEST_API. It should be purple like other macros like UCLASS.

![image](https://user-images.githubusercontent.com/33880640/36551996-789f9b8c-17be-11e8-85c8-e456b2e8e858.png)

**Comment by CJRobertson** [comment]
That was totally my mistake due to misreading. That pretty much fixed everything. I was left with one more issue which was the binding of the microphone key.

![capture](https://user-images.githubusercontent.com/13307644/36571178-49e54e78-17ea-11e8-8df8-448a8f25f281.PNG)

I got this one figured out though. In the MyPawn.cpp code example you will need to include one more header file which is `#include "Components/InputComponent.h"` Adding that header file should take care of the binding issue. I submitted a pull request for this.

![capture2](https://user-images.githubusercontent.com/13307644/36571268-c401af3a-17ea-11e8-98f0-c7ee91303558.PNG)

Thank you for the help. I really appreciate it.
